src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -33,4 +34,21 @@ public boolean deletePurchaseDetails(UUID purchaseId) throws NoRecordFoundExcept
         }
         return success;
     }
+
+    /**
+     * If the PurchaseId exists then update the PurchaseId or else send Id is mismatched.
+     *
+     * @param purchaseId
+     * @return
+     * @throws NoRecordFoundException
+     */
+    public Purchase updateUserPartially(UUID purchaseId, Purchase purchaseDetails) throws NoRecordFoundException {
+        Purchase purchase = purchaseRepository.findById(purchaseId).orElseThrow(() -> new NoRecordFoundException("User not found on :: " + purchaseId));
+        purchase.setUserName(purchaseDetails.getUserName());
```

Comment:
add validation.
throw an error if username  does not exists in user table

Reply:
Updated.